astropy/coordinates/sky_coordinate.py

```diff
@@ -2070,3 +2070,14 @@ def from_name(cls, name, frame="icrs", parse=False, cache=True):
             return icrs_sky_coord
         else:
             return icrs_sky_coord.transform_to(frame)
+
+
+@replace.register
+def _replace_SkyCoord(obj: SkyCoord, /, copy: bool = False, **kwargs: Any) -> SkyCoord:
+    """
+    A helper function for `~astropy.coordinates.BaseFrame.replace` that
+    handles the `~astropy.coordinates.SkyCoord` class.
+    """
+    # Currently ``replicate`` is not implemented for SkyCoord and returns a
+    # ``BaseCoordinateFrame``object.
+    return obj.__class__(obj.replicate(copy=copy, **kwargs))
```

Comment:
Realize this is still a draft just to see whether we want it at all, but this won't be quite right, since all non-frame attributes get lost.